routers/web/auth/oauth2_provider.go

```diff
@@ -104,7 +104,19 @@ func InfoOAuth(ctx *context.Context) {
 		Picture:  ctx.Doer.AvatarLink(ctx),
 	}
 
-	groups, err := oauth2_provider.GetOAuthGroupsForUser(ctx, ctx.Doer)
+	// groups, err := oauth2_provider.GetOAuthGroupsForUser(ctx, ctx.Doer)
+	var token string
+	if auHead := ctx.Req.Header.Get("Authorization"); auHead != "" {
```

Comment:
This code looks unrelated? This is a web router, I don't think there is a header `Authorization` will be sent.

Reply:
This is called from the API route `/login/oauth/userinfo` and it's the only API endpoint other than the response coming embedded in the `id_token` if the scope/claim comes as `groups`.

This was my attempt to get the token so I can check if the `public-only` scope is present.

In my tests, the token is captured inside the header if one performs a request like this:

```
curl -H "Authorization: Bearer **token**" http://gitea.local/login/oauth/userinfo
```

(Given that membership in the public group is made visible)

If there's more idiomatic way to capture the token inside `/web/auth/*` I would be happy to change that into more idiomatic.

Reply:
i found the better approach:

`onlyPublicGroups := ctx.IsSigned && (ctx.FormString("private") == "" || ctx.FormBool("private"))`